### PR TITLE
Test alignment decision - [MOD-6998]

### PR DESCRIPTION
**Pull request by GuyAv46**
**Describe the changes in the pull request**

Add tests for the alignment hint calculation (both logic [unit] and when creating an index [flow])

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/VecSim/spaces/spaces.cpp [hunk 1]
```diff
@@ -24,9 +24,9 @@ dist_func_t<float> GetDistFunc<vecsim_types::bfloat16, float>(VecSimMetric metri
     switch (metric) {
     case VecSimMetric_Cosine:
     case VecSimMetric_IP:
-        return IP_BF16_GetDistFunc(dim, nullptr, alignment);
+        return IP_BF16_GetDistFunc(dim, alignment);
```
Comment (meiravgri) [inline src/VecSim/spaces/spaces.cpp:27]: noice

File: tests/unit/test_spaces.cpp [hunk 2]
```diff
@@ -315,36 +328,50 @@ TEST_P(FP32SpacesOptimizationTest, FP32InnerProductTest) {
         v2[i] = (float)(i + 1.5);
     }
 
+    auto expected_alignment = [](size_t reg_bit_size, size_t dim) {
+        size_t elements_in_reg = reg_bit_size / sizeof(float) / 8;
+        return (dim % elements_in_reg == 0) ? elements_in_reg * sizeof(float) : 0;
+    };
```
Comment (meiravgri) [inline tests/unit/test_spaces.cpp:334]: The alignment is actually independent of the type, no? 
`alignment = reg_bit_size / 8;`

**Review by meiravgri — COMMENTED** [review]
Clean as always.
one lambda declaration is enough as the alignment can be concluded from the register size.

File: tests/unit/test_spaces.cpp (hunk 2 above)
Comment (GuyAv46) [reply tests/unit/test_spaces.cpp:334]: The alignment itself yes, but we only align if the `dim * sizeof(element)` is a multiplication of the register byte size

Comment (meiravgri) [reply tests/unit/test_spaces.cpp:334]: sure, i just meant that there is no need for a dedicated lambda for each type

**Review by meiravgri — APPROVED** [review]

**Pull request merged**